.github/workflows/pre-merge.yml

```diff
@@ -0,0 +1,67 @@
+name: Pre Merge Checks
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  test:
+    runs-on: [ubuntu-latest]
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Cache Gradle Folders
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches/
+            ~/.gradle/wrapper/
+          key: cache-gradle-${{ hashFiles('gradle/wrapper/gradle-wrapper.properties') }}
```

Comment:
Is that useful? To use the hash of the properties for the cache?
It seems to me that this file does not change often, meaning having a simple key works too. Worse case scenario, it pull the cache, that contains old stuff, and has to redownload things.
Unless it is here to ensure that said cache don't become bloated! Which could happen I guess if you start to store in the cache every single cache for every gradle versions....

Reply:
> Is that useful? To use the hash of the properties for the cache?

That's the file you touch every time you bump Gradle version. So it's not changing that often, but it definitely changes once you bump Gradle.

Given that a bump of Gradle requires you to download a new binary, it's reasonable to consider that a valid cache key imho :)